Sources/Workspace/UserToolchain.swift

```diff
@@ -223,10 +223,14 @@ public final class UserToolchain: Toolchain {
     }
 
     public static func deriveSwiftCFlags(triple: Triple, destination: Destination) -> [String] {
-      return (triple.isDarwin() || triple.isAndroid()
+      var flags = (triple.isDarwin() || triple.isAndroid()
         ? ["-sdk", destination.sdk.pathString]
         : [])
-        + destination.extraSwiftCFlags
+      if triple.isAndroid() {
+        let resourceDirectory = destination.sdk.appending(components: "usr", "lib", "swift")
+        flags += ["-resource-dir", resourceDirectory.pathString]
```

Comment:
hmm, this should be handled in the swift-driver I think.

cc @brentdax

Reply:
@brentdax, where does this belong? (This is the 7‐day ping.)

Reply:
Sorry, @SDGGiesbrecht. Ankit is right—if possible, the driver should determine the correct resource directory.

However, the resource directory is meant to be matched to the compiler, not the SDK—it contains things like headers for statically-linked runtime libraries. What do you have in `usr/lib/swift` of your SDK? Is it specific to the compiler you're building with, or the OS version you're building for?

Reply:
> What do you have in usr/lib/swift of your SDK?

The SDK is one of @compnerd’s artifacts from [here](https://compnerd.visualstudio.com/swift-build/_build/results?buildId=27980&view=artifacts&type=publishedArtifacts). When I look inside `usr/lib/swift`, there is an `android` directory with the `.swiftmodule` and `.so` files for the standard library, Foundation, Dispatch and Glibc, as well as a bunch of C headers named `ThisShim.h` and `ThatOverlayShims.h`.

The (first) error I get without this flag is:

> <unknown>:0: error: cannot load underlying module for 'Dispatch'

(See the links at the top of the PR for complete context.)

> Is it specific to the compiler you're building with, or the OS version you're building for?

It seems it basically _is_ the SDK. The only things not in that subdirectory are licences and documentation—`.txt`, `.md`, `.vim`, etc.

> if possible, the driver should determine the correct resource directory.

If I should send a PR to put this in the driver, can you point me to where approximately I should put it?

Reply:
@brentdax, still waiting for an answer about were to put this instead.